### ISSUE TITLE
[built-ins] Error on activation: markdown-language-features and image-preview

### ISSUE THREAD

**Issue by marcdumais-work** [eclipse-theia/vscode-builtin-extensions#14]
These two built-ins seem to have the same problem: 

_Error on activation of markdown-language-features TypeError: n.window.registerWebviewEditorProvider is not a function_

_Error on activation of image-preview TypeError: s.window.registerWebviewEditorProvider is not a function_

Searching for `registerWebviewEditorProvider` in Theia, I came-up empty, which could explain why it's not found during activation. 

Opening one of the places it's used in VS Code, I found the following (see screen capture), that seem to indicate that this function is part of an experimental API and should not be used in published extensions. I guess their own built-ins are the beta users for such features. Do we need to support the experimental API? Or perhaps use an older version of these extensions, that was not yet using this API?

![image](https://user-images.githubusercontent.com/25749063/70245518-343f4c80-1744-11ea-93e5-3443109399d4.png)

Full error traces:
```
root ERROR [hosted-plugin: 20436] Error on activation of markdown-language-features TypeError: n.window.registerWebviewEditorProvider is not a function
    at new c (/tmp/vscode-unpacked/markdown-language-features-0.3.0-next.9e899b556f.vsix/extension/dist/extension.js:1:67604)
    at t.activate (/tmp/vscode-unpacked/markdown-language-features-0.3.0-next.9e899b556f.vsix/extension/dist/extension.js:1:50702)
    at PluginManagerExtImpl.<anonymous> (/tmp/a/node_modules/@theia/plugin-ext/lib/plugin/plugin-manager.js:461:87)
    at step (/tmp/a/node_modules/@theia/plugin-ext/lib/plugin/plugin-manager.js:47:23)
    at Object.next (/tmp/a/node_modules/@theia/plugin-ext/lib/plugin/plugin-manager.js:28:53)
    at fulfilled (/tmp/a/node_modules/@theia/plugin-ext/lib/plugin/plugin-manager.js:19:58)
root INFO [hosted-plugin: 20436] PLUGIN_HOST(20436): PluginManagerExtImpl/loadPlugin(/tmp/vscode-unpacked/image-preview-0.3.0-next.9e899b556f.vsix/extension/dist/extension)
root ERROR [hosted-plugin: 20436] Error on activation of image-preview TypeError: s.window.registerWebviewEditorProvider is not a function
    at t.activate (/tmp/vscode-unpacked/image-preview-0.3.0-next.9e899b556f.vsix/extension/dist/extension.js:1:7708)
    at PluginManagerExtImpl.<anonymous> (/tmp/a/node_modules/@theia/plugin-ext/lib/plugin/plugin-manager.js:461:87)
    at step (/tmp/a/node_modules/@theia/plugin-ext/lib/plugin/plugin-manager.js:47:23)
    at Object.next (/tmp/a/node_modules/@theia/plugin-ext/lib/plugin/plugin-manager.js:28:53)
    at fulfilled (/tmp/a/node_modules/@theia/plugin-ext/lib/plugin/plugin-manager.js:19:58)
```
BTW, these seem to be recent changes:

image-preview:
```
commit aaa01eac73a265fcd190403040ed34a67839d773
Author: Matt Bierner <matb@microsoft.com>
Date:   Thu Nov 21 20:36:39 2019 -0800

    Make previewManager implement WebviewEditorProvider directly

```

markdown-language-features:
```
commit 8236f06bfc4c20318b6bd4b9a8823f565a6b2cc0
Author: Matt Bierner <matb@microsoft.com>
Date:   Thu Nov 7 23:17:35 2019 -0800

    Add experimental webview editor based markdown preview
    
    Potentially for #54776, #2766

```

__Edit: actually this new API was added to image-preview a bit earlier:__

```
commit 011836a150c5bb55a45ab03896518e159396e031 (HEAD)
Author: Matt Bierner <matb@microsoft.com>
Date:   Tue Sep 10 17:56:57 2019 -0700

    Prototyping custom editors (#77789)
    
    * Custom Editor exploration
    
    For #77131
    
    Adds a prototype of custom editors contributed by extensions. This change does the following:
    
    - Introduces a new contribution point for the declarative parts of a custom editor
    - Adds API for registering a webview editor provider. This lets VS Code decided when to create a webview editor
    - Adds an `openWith` command that lets you select which editor to use to open a resource from the file explorer
    - Adds a setting that lets you say that you always want to use a custom editor for a given file extension
    - Hooks up auto opening of a custom editor when opening a file from quick open or explorer
    - Adds a new extension that contributes a custom image preview for png and jpg files
[...]
```

**Comment by marcdumais-work** [comment]
cc: @tsmaeder @benoitf

**Comment by vince-fugnitto** [comment]
@marcdumais-work I believe this is the relevant issue in Theia:
https://github.com/eclipse-theia/theia/issues/6636

**Comment by marcdumais-work** [comment]
Thanks @vince-fugnitto - I see that Anton already answered my main questions. 

I'm trying to go back a bit in VS Code history to see if I can at least get markdown-language-support working.

**Comment by vince-fugnitto** [comment]
> I'm trying to go back a bit in VS Code history to see if I can at least get markdown-language-support working.

Perhaps it is enough to simply use `@theia/vscode-builtin-markdown` and not the `language-features`? Looking at the code, the former simply contributes the preview for markdown, something we already handle so perhaps it is not necessary at the moment?

**Comment by marcdumais-work** [comment]
> I'm trying to go back a bit in VS Code history to see if I can at least get markdown-language-support working.

Markdown built-in works with vscode version 1.40.0

**Comment by vince-fugnitto** [comment]
> > I'm trying to go back a bit in VS Code history to see if I can at least get markdown-language-support working.
> 
> Markdown built-in works with vscode version 1.40.0

Should we try to publish different versions of the builtins based on their VS Code version?
For example: `@theia/vscode-builtin-java:1.40.0` ?

**Comment by benoitf** [comment]
would be nice to have github releases for each stable versions yes

**Comment by marcdumais-work** [comment]
The working markdown-language-features extension we had been using so far, v1.39.2, I think I had manually built/published a while ago, is gone from open-vsx.org (cleanup of extensions published by non-publishers, as per the new rules). That's fine - I think we should re-publish it using this repo's CI, like we automatically do nightly for solid and preview versions of the built-ins. 

I have recently added a couple of GH workflows that permit doing this for an arbitrary version of vscode, permitting e.g. to publish versions of builtins that might be missing (solid or preview).
